components/search/FacetDropdown.vue

```diff
@@ -0,0 +1,248 @@
+        @click.stop="applySelection"
+      >
+        {{ $t('facets.button.apply') }}
+      </b-button>
+    </li>
+
+    <li
+      v-else
+      class="p-2 float-right"
+    >
+      <b-button
+        variant="link"
+        :disabled="disableRadioResetButton"
+        @click.stop="resetRadioSelection"
+      >
+        {{ $t('facets.button.reset') }}
+      </b-button>
+    </li>
+  </b-dropdown>
+</template>
+
+<script>
+  import vClickOutside from 'v-click-outside';
+
+  export default {
+    directives: {
+      clickOutside: vClickOutside.directive
+    },
+
+    props: {
+      facet: {
+        type: Object,
+        required: true
+      },
+
+      selectedFacet: {
+        type: [Array, String],
+        required: false,
+        default: () => []
+      },
+
+      facetType: {
+        type: String,
+        required: true
+      }
+    },
+
+    data() {
+      return {
+        RADIO: 'radio',
+        CHECKBOX: 'checkbox',
+        THEME: 'THEME',
+        preSelected: [],
+        radioSelected: null
+      };
+    },
+
+    computed: {
+      sortOptions() {
+        let newArray = [].concat(this.facet.fields);
+        const selected = [];
+
+        /* TODO: THIS NEEDS TO BE CLEANED UP */
+        newArray.map(field => {
+          if (this.selectedFacet.includes(field.label)) {
+            selected.push(field);
+          }
+        });
+
+        const leftOver = newArray.filter(field => !this.selectedFacet.includes(field.label));
+
+        return selected.sort((a, b) => a.count + b.count).concat(leftOver);
+        /* END TODO */
+      },
+
+      facetName() {
+        return this.$t(`facets.${this.facet.name}.name`);
+      },
+
+      isRadio() {
+        return this.facetType === this.RADIO;
+      },
+
+      activateCheckboxResetButton() {
+        return this.preSelected.length > 0;
+      },
+
+      disableRadioResetButton() {
+        return !this.radioSelected;
+      },
+
+      disableApplyButton() {
+        return this.preSelected.length === this.selectedFacet.length;
+      },
+
+      dropdownVariant() {
+        return (this.radioSelected || this.selectedFacet.length > 0) ? 'secondary' : 'light';
+      }
+    },
+
+    mounted() {
+      if (this.isRadio) {
+        if (Array.isArray(this.selectedFacet)) {
+          this.radioSelected = '';
+        } else {
+          this.radioSelected = this.selectedFacet;
+        }
+      } else if (this.selectedFacet.length > 0) {
+        this.preSelected = this.selectedFacet;
+      }
+    },
+
+    methods: {
+      resetCheckboxSelection() {
+        this.preSelected = [];
+      },
+
+      resetRadioSelection() {
+        this.radioSelected = '';
+      },
+
+      idGenerator(label, type) {
+        return label.replace(/\s+/g, '_').toLowerCase() + '_' + type;
+      },
+
+      applySelection() {
+        if (this.isRadio) {
+          this.$emit('updated', this.THEME, this.radioSelected);
+          this.$refs.dropdown.hide(true);
```

Comment:
This `this.$refs.dropdown.hide(true);` is the same in the if as well as in the else. I guess, it can go outside of the if/else statement and have it just once.